buildSrc/src/main/java/org/elasticsearch/gradle/testclusters/ElasticsearchNode.java

```diff
@@ -59,22 +102,282 @@ public Distribution getDistribution() {
+            }
+        });
+    }
+
+    synchronized void start() {
         logger.info("Starting `{}`", this);
+
+        File distroArtifact = new File(
+            new File(artifactsExtractDir, distribution.getFileExtension()),
+            distribution.getFileName() + "-" + getVersion()
+        );
+        if (distroArtifact.exists() == false) {
+            throw new TestClustersException("Can not start " + this + ", missing: " + distroArtifact);
+        }
+        if (distroArtifact.isDirectory() == false) {
+            throw new TestClustersException("Can not start " + this + ", is not a directory: " + distroArtifact);
+        }
+        services.sync(spec -> {
+            spec.from(new File(distroArtifact, "config"));
+            spec.into(getConfigFile().getParent());
+        });
+        writeConfigFile();
+        startElasticsearchProcess(distroArtifact);
+        registerCleanupHooks();
     }
 
-    void stop(boolean tailLogs) {
+    private void startElasticsearchProcess(File distroArtifact) {
+        logger.info("Running `bin/elasticsearch` in `{}` for {}", workingDir, this);
+        final ProcessBuilder processBuilder = new ProcessBuilder();
+        if (OperatingSystem.current().isWindows()) {
+            processBuilder.command(
+                "cmd", "/c",
+                new File(distroArtifact, "\\bin\\elasticsearch.bat").getAbsolutePath()
+            );
+        } else {
+            processBuilder.command(
+                new File(distroArtifact.getAbsolutePath(), "bin/elasticsearch").getAbsolutePath()
+            );
+        }
+        try {
+            processBuilder.directory(workingDir);
+            Map<String, String> environment = processBuilder.environment();
+            // Don't inherit anything from the environment for as that would  lack reproductability
+            environment.clear();
+            if (javaHome != null) {
+                environment.put("JAVA_HOME", getJavaHome().getAbsolutePath());
+            } else if (System.getenv().get("JAVA_HOME") != null) {
+                logger.warn("{}: No java home configured will use it from environment: {}",
+                    this, System.getenv().get("JAVA_HOME")
+                );
+                environment.put("JAVA_HOME", System.getenv().get("JAVA_HOME"));
+            } else {
+                logger.warn("{}: No javaHome configured, will rely on default java detection", this);
+            }
+            environment.put("ES_PATH_CONF", getConfigFile().getParentFile().getAbsolutePath());
+            environment.put("ES_JAVA_OPTIONS", "-Xms512m -Xmx512m");
+            // don't buffer all in memory, make sure we don't block on the default pipes
+            processBuilder.redirectError(ProcessBuilder.Redirect.appendTo(getStdErrFile()));
+            processBuilder.redirectOutput(ProcessBuilder.Redirect.appendTo(getStdoutFile()));
+            esProcess = processBuilder.start();
```

Comment:
It's not because we do our own cleanup.
It's hard to write a test that would assert that no ES processes are left after executing,
maybe we could do it in an after build hook that always runs if we really wanted to, but  I always check for them when developing and I haven't found any with this implementation so far. 
I have ran into cases where the ant runner left processes behind too.

Reply:
We don't have a way to easily test this right now. Will consider adding something lather on if this turns out to be problematic.